### PR TITLE
Added token-based authentication.

### PR DESCRIPTION
**Pull request by MorganEPatch**
Public API for token-based authentication added.

### CODE REVIEW

File: source/draft_auth.rst [hunk 1]
```diff
@@ -33,6 +33,30 @@ fields as necessary for that kind of connection.
 
 These other fields might be things like ``password``, ``token``, ``key``, etc.
 
+Token-based authentication
+--------------------------
+
+Token-based is the primary and default authentication method for TimeSync. TimeSync
+provides an endpoint at ``/login``, which should be sent a POST request, with the body
+fitting one of the other available authentication schemes (e.g. Password, LDAP). The
+endpoint will then return a JWT token string as its response body. This response body
```
Comment (pop) [inline source/draft_auth.rst]: JWT? Define yo' Three Letter Acronyms (TLAs).

File: source/draft_auth.rst [hunk 2]
```diff
@@ -33,6 +33,30 @@ fields as necessary for that kind of connection.
...
+POST endpoints:
+
+.. code-block:: javascript
+
+    {
+      "auth": {
+        "type": "token",
+        "token": ...
+      },
+      ...
```
Comment (ElijahCaine) [inline source/draft_auth.rst]: You should add a little more context or a complete example below (hitting /times/:some_uuid you would craft this JSON body...).

**Comment by Kennric** [comment]
I agree with @ElijahCaine 's comments, and also think it would be a good idea to link directly to the docs on specific auth schemes where you mention them (ldap, password, etc). Full examples would definitely help.

**Comment by pop** [comment]
I agree with @Kennric, add an LDAP auth section (similar to the password auth section) and perhaps a toctree (`.. contents::` below the introduction paragraph).

**Comment by matthewrsj** [comment]
+1 on the content. The examples for the token auth are really easy to follow.

**Comment by Kennric** [comment]
+1

**Comment by pop** [comment]
lgtm, I agree that the examples are very easy to follow. +1

**Pull request merged**